oe_theme.theme

```diff
@@ -210,153 +210,300 @@ function oe_theme_preprocess_paragraph__oe_links_block(array &$variables): void
  * Implements hook_preprocess_oe_corporate_blocks_ec_footer().
  */
 function oe_theme_preprocess_oe_corporate_blocks_ec_footer(array &$variables): void {
+  $variables['processed'] = [];
   $corporate_footer = $variables['corporate_footer'];
-
-  // Create the section for 'About EC'.
-  $about_ec = $corporate_footer['about_ec'];
-  $sections_about_ec = [
-    'title' => $about_ec['title'],
-    'links' => [],
+  $site_name = $corporate_footer['site_name'];
+  $ec_site_link = [
+    'link' => [
+      'label' => $site_name['label'],
+      'path' => $site_name['href'],
+    ],
   ];
 
-  foreach ($about_ec['items'] as $item) {
-    $sections_about_ec['links'][] = [
+  $variables['processed']['site_name'] = \Drupal::config('system.site')->get('name');
+  $variables['processed']['site_path'] = Url::fromRoute('<front>', [], ['absolute' => TRUE])->toString();
+  $variables['processed']['ec_site_link'] = $ec_site_link;
+  $variables['processed']['site_owner'] = t('This site is managed by the @name', ['@name' => $variables['site_owner']]);
+
+  $variables['processed']['class_navigation'] = _oe_theme_preprocess_footer_links($corporate_footer['class_navigation']);
+  $variables['processed']['legal_navigation'] = _oe_theme_preprocess_footer_links($corporate_footer['legal_navigation']);
+
+  $service_navigation = [];
+  foreach ($corporate_footer['service_navigation'] as $link) {
+    $service_navigation_link = [
       'link' => [
-        'label' => $item['label'],
-        'path' => $item['href'],
+        'label' => $link['label'],
+        'path' => $link['href'],
         'icon_position' => 'after',
       ],
     ];
-  }
-
-  // Create the section for 'Social media'.
-  $social_media = $corporate_footer['social_media'];
-  $sections_social_media = [
-    'title' => $social_media['title'],
-    'links' => [],
-  ];
-
-  foreach ($social_media['items'] as $item) {
-    if ($item['type'] == 'social-network') {
-      $sections_social_media['links'][] = [
-        'link' => [
-          'label' => $item['link']['label'],
-          'path' => $item['link']['href'],
-          'icon_position' => 'before',
-        ],
-        'icon' => [
-          'path' => $variables['ecl_icon_path'],
-          'type' => 'branded',
-          'name' => $item['icon'],
-        ],
-      ];
-    }
-    else {
-      $sections_social_media['links'][] = [
-        'link' => [
-          'label' => $item['link']['label'],
-          'path' => $item['link']['href'],
-          'icon_position' => 'after',
-        ],
+    if (!empty($link['external']) && $link['external'] === TRUE) {
+      $service_navigation_link += [
         'icon' => [
           'path' => $variables['ecl_icon_path'],
           'type' => 'ui',
           'name' => 'external',
         ],
       ];
     }
+    $service_navigation[] = $service_navigation_link;
   }
+  $variables['processed']['service_navigation'] = $service_navigation;
+
+  $ecl_branding = theme_get_setting('branding') ?? 'core';
```

Comment:
changed, but I have a concern, (the reason why I did it like this in the first place) I tested the upgrade from 2.x branch to this one, and it could not find the right template.
Granted the code changed quite a bit since then, but I was not able to retest this scenario, I tried but updb was with errors:
`root@dbe60b6d5f48:/var/www/html# ./vendor/bin/drush updb -y
 [error]   (Currently using Missing or invalid module Array) `
`[notice] Module file_link has an entry in the system.schema key/value storage, but is missing from your site. <a href=":url">More information about this error</a>.
>  [notice] Update started: oe_corporate_blocks_post_update_30001
>  [error]  The module file_link does not exist. 
>  [error]  Update failed: oe_corporate_blocks_post_update_30001 
 [error]  Update aborted by: oe_corporate_blocks_post_update_30001 
 [error]  Finished performing updates. `

maybe I did something wrong with my upgrade procedure, please confirm if this works for you.